gcc/rust/expand/rust-macro-builtins-asm.cc

```diff
@@ -16,5 +16,709 @@
 // along with GCC; see the file COPYING3.  If not see
 // <http://www.gnu.org/licenses/>.
 
-#include "rust-macro-builtins.h"
-#include "rust-macro-builtins-helpers.h"
+#include "rust-macro-builtins-asm.h"
+
+namespace Rust {
+std::map<AST::InlineAsmOption, std::string> InlineAsmOptionMap{
+  {AST::InlineAsmOption::PURE, "pure"},
+  {AST::InlineAsmOption::NOMEM, "nomem"},
+  {AST::InlineAsmOption::READONLY, "readonly"},
+  {AST::InlineAsmOption::PRESERVES_FLAGS, "preserves_flags"},
+  {AST::InlineAsmOption::NORETURN, "noreturn"},
+  {AST::InlineAsmOption::NOSTACK, "nostack"},
+  {AST::InlineAsmOption::MAY_UNWIND, "may_unwind"},
+  {AST::InlineAsmOption::ATT_SYNTAX, "att_syntax"},
+  {AST::InlineAsmOption::RAW, "raw"},
+};
+
+tl::expected<InlineAsmContext, std::string>
+parse_clobber_abi (InlineAsmContext inline_asm_ctx)
```

Comment:
this is good! in another PR, I'd like you to start working on a "proper" error type instead of strings, something like

```c++
enum class AsmParseError {
    Reason1,
    Reason2,
    MissingAbiArgument,
    ...
}
```

it'll clean up the code a little bit and we can then see how we'll transform these into proper `Error`s to emit for the user